packages/editor-ui/src/modules/executions.ts

```diff
@@ -0,0 +1,125 @@
+		},
+		executionTimeout: (state): number => {
+			return state.executionTimeout;
+		},
+		maxExecutionTimeout: (state): number => {
+			return state.maxExecutionTimeout;
+		},
+		// Active Execution
+		executingNode: (state): string | null => {
+			return state.executingNode;
+		},
+		activeExecutionId: (state): string | null => {
+			return state.executionId;
+		},
+		executionWaitingForWebhook: (state): boolean => {
+			return state.executionWaitingForWebhook;
+		},
+	},
+	mutations: {
+		// Active Executions
+		addActiveExecution(state, newActiveExecution: IExecutionsCurrentSummaryExtended) {
+			// Check if the execution exists already
+			const activeExecution = state.activeExecutions.find(execution => {
+				return execution.id === newActiveExecution.id;
+			});
+
+			if (activeExecution !== undefined) {
+				// Exists already so no need to add it again
+				if (activeExecution.workflowName === undefined) {
+					activeExecution.workflowName = newActiveExecution.workflowName;
+				}
+				return;
+			}
+
+			state.activeExecutions.unshift(newActiveExecution);
+		},
+		finishActiveExecution(state, finishedActiveExecution: IPushDataExecutionFinished) {
+			// Find the execution to set to finished
+			const activeExecution = state.activeExecutions.find(execution => {
+				return execution.id === finishedActiveExecution.executionId;
+			});
+
+			if (activeExecution === undefined) {
+				// The execution could not be found
+				return;
+			}
+
+			if (finishedActiveExecution.executionId !== undefined) {
+				Vue.set(activeExecution, 'id', finishedActiveExecution.executionId);
+			}
+
+			Vue.set(activeExecution, 'finished', finishedActiveExecution.data.finished);
+			Vue.set(activeExecution, 'stoppedAt', finishedActiveExecution.data.stoppedAt);
+		},
+		setActiveExecutions(state, newActiveExecutions: IExecutionsCurrentSummaryExtended[]) {
+			Vue.set(state, 'activeExecutions', newActiveExecutions);
+		},
+		setExecutingNode(state, executingNode: string) {
+			state.executingNode = executingNode;
+		},
+		setExecutionWaitingForWebhook(state, newWaiting: boolean) {
+			state.executionWaitingForWebhook = newWaiting;
+		},
+		setActiveExecutionId(state, executionId: string | null) {
+			state.executionId = executionId;
+		},
+		setSaveDataErrorExecution(state, newValue: string) {
+			Vue.set(state, 'saveDataErrorExecution', newValue);
+		},
+		setSaveDataSuccessExecution(state, newValue: string) {
+			Vue.set(state, 'saveDataSuccessExecution', newValue);
+		},
+		setSaveManualExecutions(state, saveManualExecutions: boolean) {
+			Vue.set(state, 'saveManualExecutions', saveManualExecutions);
+		},
+		setTimezone(state, timezone: string) {
+			Vue.set(state, 'timezone', timezone);
+		},
+		setExecutionTimeout(state, executionTimeout: number) {
+			Vue.set(state, 'executionTimeout', executionTimeout);
+		},
+		setMaxExecutionTimeout(state, maxExecutionTimeout: number) {
+			Vue.set(state, 'maxExecutionTimeout', maxExecutionTimeout);
+		},
+	},
```

Comment:
this is missing dispatches... we want to also move the api fetching logic here... we want to separate businness logic from rending logic as much as possible...